framework/src/media/media_player.c

```diff
@@ -150,9 +150,12 @@ play_result_t media_stop_play(void)
 	}
 
 	g_pc.state = PLAY_STOPPING;
+	PLAY_UNLOCK();
+
 	pthread_join(g_pc.pth, NULL);
+	
+	PLAY_LOCK();
```

Comment:
Why is it different in player and recorder ? You are locking and unlocking again in media_stop_play  to set pth variable to NULL but not in media_stop_record.